scripts/genbank_cds.py

```diff
@@ -0,0 +1,135 @@
+            db_, acc_, *_ = name.split("_")
+            if db_ == db and acc_.startswith(acc):
+                full_name = name
+                break
+
+    if full_name is None:
+        return None
+    else:
+        url = "htt" + url[3:]
+        return (
+            f"{url}/{full_name}/{full_name}_genomic.fna.gz",
+            f"{url}/{full_name}/{full_name}_genomic.gff.gz",
+            f"{url}/{full_name}/{full_name}_cds_from_genomic.fna.gz",
+            f"{url}/{full_name}/{full_name}_assembly_report.txt",
+        )
+
+
+def get_taxid_from_assembly_report(url):
+    print(f"opening assembly report: {url}", file=sys.stderr)
+    with urllib.request.urlopen(url) as response:
+        content = response.read()
+    print("done!", file=sys.stderr)
+
+    content = content.decode("utf-8").splitlines()
+    for line in content:
+        if "Taxid:" in line:
+            line = line.strip()
+            pos = line.find("Taxid:")
+            assert pos >= 0
+            pos += len("Taxid:")
+            taxid = line[pos:]
+            taxid = taxid.strip()
+            return taxid
+
+    assert 0
+
+
+def get_tax_name_for_taxid(taxid):
+    tax_url = (
+        f"https://www.ncbi.nlm.nih.gov/taxonomy/?term={taxid}&report=taxon&format=text"
+    )
+    print(f"opening tax url: {tax_url}", file=sys.stderr)
+    with urllib.request.urlopen(tax_url) as response:
+        content = response.read()
+
+    print("done!", file=sys.stderr)
+
+    root = etree.fromstring(content)
+    notags = etree.tostring(root).decode("utf-8")
+    if notags.startswith("<pre>"):
+        notags = notags[5:]
+    if notags.endswith("</pre>"):
+        notags = notags[:-6]
+    notags = notags.strip()
+
+    return notags
+
+
+def main(args):
+    fieldnames = ["acc", "genome_url", "gff_url", "cds_url", "assembly_report_url", "ncbi_tax_name", "ncbi_taxid"]
+    if args.output:
+        fp = open(args.output, "wt")
+        w = csv.DictWriter(fp, fieldnames=fieldnames)
+    else:
+        w = csv.DictWriter(sys.stdout, fieldnames=fieldnames)
+    w.writeheader()
+
+    acc = args.accession
+
+    genome_url, gff_url, cds_url, assembly_report_url = url_for_accession(acc)
+    taxid = get_taxid_from_assembly_report(assembly_report_url)
+    tax_name = get_tax_name_for_taxid(taxid)
+
+    d = dict(
+        acc=acc,
+        genome_url=genome_url,
+        gff_url=gff_url,
+        cds_url=cds_url,
+        assembly_report_url=assembly_report_url,
+        ncbi_tax_name=tax_name,
+        ncbi_taxid=taxid,
+    )
+
+    w.writerow(d)
+    print(f"retrieved for {acc} - {tax_name}", file=sys.stderr)
+    if fp:
+        fp.close()
+
+    return 0
+
+def cmdline(sys_args):
+    "Command line entry point w/argparse action."
+    p = argparse.ArgumentParser()
+    p.add_argument("accession")
+    p.add_argument("-o", "--output")
+    args = p.parse_args()
+    return main(args)
+
+if __name__ == '__main__':
+    returncode = cmdline(sys.argv[1:])
+    sys.exit(returncode)
```

Comment:
This is something I inherited from DIB-lab, but I think the reasoning is:
`cmdline` is a function that returns a 0 (success) or non-zero, error, or None (implied success)
Then `sys.exit` takes that and makes it the script return code
So `sys.argv` is a way for the command line arguments to be passed into `cmdline`
I think it's split out for code that usually becomes integrated into software (as opposed to a repo like this that is more for data analysis), because it helps with unit testing. 
`cmdline` is a function that can be tested, which is one reason it’s been split out